cli/tests/services.bats

```diff
@@ -978,6 +978,64 @@ EOF
   assert_line "inner _FLOX_SERVICES_TO_START=unset"
 }
 
+@test "activate: services can be layered" {
+
+  MANIFEST_CONTENTS_1="$(cat << "EOF"
+    version = 1
+
+    [services]
+    one.command = "echo one"
+EOF
+  )"
+
+  "$FLOX_BIN" init -d one
+  echo "$MANIFEST_CONTENTS_1" | "$FLOX_BIN" edit -d one -f -
+
+
+  MANIFEST_CONTENTS_2="$(cat << "EOF"
+    version = 1
+
+    [services]
+    two.command = "echo two"
+EOF
+  )"
+
+  "$FLOX_BIN" init -d two
+  echo "$MANIFEST_CONTENTS_2" | "$FLOX_BIN" edit -d two -f -
+
+  cat <<"EOF" > script_inner.sh
+    set -euo pipefail
+
+    for i in {1..5}; do
+      if "$FLOX_BIN" services status -d two | grep "two.*Completed"; then
+        break
+      fi
+      sleep .1
+    done
+    if [ "$i" -eq 5 ]; then
+      echo "Service in inner activation did not run" >&3
+      exit 1
+    fi
+EOF
+
+  "$FLOX_BIN" activate -d one --start-services -- bash <(cat <<"EOF"
+    set -euo pipefail
```

Comment:
blocking: This new test was hanging for me locally. I figured out why and created https://github.com/flox/flox/issues/2002 for the underlying problem, but for now we can plaster over it with:
```suggestion
  cat <<"EOF" > script_inner.sh
    source "${TESTS_DIR}/services/register_cleanup.sh"

    for i in {1..5}; do
      if "$FLOX_BIN" services status -d two | grep "two.*Completed"; then
        break
      fi
      sleep .1
    done
    if [ "$i" -eq 5 ]; then
      echo "Service in inner activation did not run" >&3
      exit 1
    fi
EOF

  "$FLOX_BIN" activate -d one --start-services -- bash <(cat <<"EOF"
    source "${TESTS_DIR}/services/register_cleanup.sh"
```

Reply:
Fixing